models/demos/t3000/llama2_70b/tests/test_llama_attention.py

```diff
@@ -315,52 +317,148 @@ def run_test_LlamaAttention_inference(
+                        rope_setup=None,
+                        use_scaled_rope=configuration.use_scaled_rope,
+                    )
+                    tt_chunk_out = tt_LlamaAttention_model(
+                        attention_input,
+                        rot_mat,
+                        None,
+                        cache_idxs=None,
+                        page_table=prefill_page_table_tt,
+                        mode=mode,
+                        chunk_page_table=chunk_page_table_tt,
+                        chunk_start_idx=chunk_start,
+                    )
+
+                    tt_chunk_out = ttnn.from_device(tt_chunk_out)
+                    tt_chunk_out = ttnn.to_torch(tt_chunk_out, mesh_composer=ConcatMeshToTensor(t3k_mesh_device, dim=3))
+                    tt_chunk_out = tt_chunk_out.permute(2, 1, 0, 3).squeeze(1)  # [batch, seq_len, hidden_dim]
+
+                    # check outputs ----------------------------------------------------------------------
+                    pytorch_chunk_out = pytorch_out[:, chunk_start:chunk_end]
+                    does_pass, output_pcc = comp_pcc(pytorch_chunk_out, tt_chunk_out, pcc)
+                    logger.info(f"Chunk {chunk_start} output: {output_pcc}")
+                    all_pccs.append(extract_pcc_from_log(output_pcc))
+
+            else:
+                # TT hardware execution -------------------------------------------------------------
+                attention_input, start_pos, rot_mat, cache_idxs = tt_llama_attention_prepare_inputs(
+                    tt_LlamaAttention_model,
+                    tt_input,
+                    start_pos,
+                    mode,
+                    configuration.rope_theta,
+                    rope_setup=None,
+                    use_scaled_rope=configuration.use_scaled_rope,
+                )
 
-        tt_out = tt_LlamaAttention_model(
-            attention_input,
-            rot_mat,
-            start_pos,
-            cache_idxs=cache_idxs if mode == "decode" else None,
-            page_table=page_table_tt,
-            mode=mode,
-        )
+                tt_out = tt_LlamaAttention_model(
+                    attention_input,
+                    rot_mat,
+                    start_pos,
+                    cache_idxs=None,
+                    page_table=page_table_tt,
+                    mode=mode,
+                )
+
+                tt_out = ttnn.from_device(tt_out)
+                tt_out = ttnn.to_torch(tt_out, mesh_composer=ConcatMeshToTensor(t3k_mesh_device, dim=3))
+                tt_out = tt_out.permute(2, 1, 0, 3).squeeze(1)  # [batch, seq_len, hidden_dim]
+
+                # check outputs ----------------------------------------------------------------------
+                does_pass, output_pcc = comp_pcc(pytorch_out, tt_out, pcc)
+                logger.info(f"Output: {output_pcc}")
+                all_pccs.append(extract_pcc_from_log(output_pcc))
+
+        else:
```

Comment:
The code in the decode path is nearly the same as the prefill (non-chunked path) right? Can most of it be shared?